### PR TITLE
Add validation for past time whene creating lesson

### PR DESCRIPTION
**Pull request by ihor-mk**

### CODE REVIEW

File: frontend/src/app/modules/lessons/lessons-create/lessons-create.component.html [hunk 1]
```diff
@@ -23,9 +23,10 @@
             </div>
 
             <mat-error class="error-container">
-                <div class="error" *ngIf="date?.invalid && submitted">
+                <div class="error" *ngIf="(date?.invalid || isPast) && submitted">
                     <img src="/assets/user-profile/alert-triangle.svg" alt="" />
-                    <p>Date is Required</p>
+                    <p *ngIf="!isPast">Date is Required</p>
+                    <p *ngIf="isPast">That time has passed</p>
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/lessons/lessons-create/lessons-create.component.html:29]: Lesson with past date can't be created

File: frontend/src/app/modules/lessons/lessons-create/lessons-create.component.html [hunk 2]
```diff
@@ -45,9 +46,10 @@
             </div>
 
             <mat-error class="error-container">
-                <div class="error" *ngIf="!time && submitted">
+                <div class="error" *ngIf="(!time || isPast) && submitted">
                     <img src="/assets/user-profile/alert-triangle.svg" alt="" />
-                    <p>Time is Required</p>
+                    <p *ngIf="!isPast">Time is Required</p>
+                    <p *ngIf="isPast">That time has passed</p>
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/lessons/lessons-create/lessons-create.component.html:52]: Selected time has passed, Select future one

File: frontend/src/app/modules/lessons/lessons-create/lessons-create.component.ts [hunk 3]
```diff
@@ -97,13 +99,21 @@ export class LessonsCreateComponent implements OnInit {
         }
 
         const [hours, minutes] = this.time.split('.');
-        const startAt = moment(this.date?.value, 'YYYY-MM-DD').set({ hour: parseInt(hours, 10), minute: parseInt(minutes, 10) }).toDate();
+        const startAt = moment(this.date?.value, 'YYYY-MM-DD')
+            .set({ hour: parseInt(hours, 10), minute: parseInt(minutes, 10) })
+            .toDate();
+
+        const timeNow = moment(Date.now()).toDate();
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/lessons/lessons-create/lessons-create.component.ts:106]: moment() is the same as moment(Date.now())

File: frontend/src/app/modules/lessons/lessons-create/lessons-create.component.ts [hunk 4]
```diff
@@ -97,13 +99,21 @@ export class LessonsCreateComponent implements OnInit {
...
+
+        this.isPast = startAt < timeNow;
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/lessons/lessons-create/lessons-create.component.ts:108]: Can you compare it via moment?

**Review by Anatolii-Ivanov — CHANGES_REQUESTED** [review]

**Review by tatianahutii — APPROVED** [review]

**Review by Anatolii-Ivanov — APPROVED** [review]

**Pull request merged**